### PR TITLE
Remove an unnecessary warning from the runner

### LINKED ISSUE #6395: Weird [INFO][WARNING] messages emitted by clone
I've recently started to see weird ` [INFO][WARNING]`` messages like this during cloning:

```
(handbook2) adina@muninn in /tmp
❱ datalad clone git@github.com:adswa/windowstesting.git mimimi
[INFO   ] Unable to parse git config from origin                                                                  
[INFO   ] [WARNING] Unknown instance class: <class '_io.BufferedRandom'>, assuming file-like input: <_io.BufferedRandom name=3> 
[INFO   ] [WARNING] Unknown instance class: <class '_io.TextIOWrapper'>, assuming file-like input: <_io.TextIOWrapper name='<stdin>' mode='r' encoding='utf-8'> 
[INFO   ] Remote origin does not have git-annex installed; setting annex-ignore 
[INFO   ] This could be a problem with the git-annex installation on the remote. Please make sure that git-annex-shell is available in PATH when you ssh into the remote. Once you have fixed the git-annex installation, run: git annex enableremote origin 
install(ok): /tmp/mimimi (dataset)
```

``debug`` says that they originate from the stderr of a git clone call:

```
[DEBUG  ] Run ['git', '-c', 'diff.ignoreSubmodules=none', 'clone', '--progress', 'git@github.com:adswa/windowstesting.git', '/tmp/mimimimi'] (cwd=None) 
[DEBUG  ] Non-progress stderr: b"Cloning into '/tmp/mimimimi'...\n"                                               
[DEBUG  ] Non-progress stderr: b"[WARNING] Unknown instance class: <class '_io.BufferedRandom'>, assuming file-like input: <_io.BufferedRandom name=3> \n" 
[DEBUG  ] Non-progress stderr: b"[WARNING] Unknown instance class: <class '_io.TextIOWrapper'>, assuming file-like input: <_io.TextIOWrapper name='<stdin>' mode='r' encoding='utf-8'> \n" 
[DEBUG  ] Non-progress stderr: b' \r'                                                                             
[DEBUG  ] Non-progress stderr: b'remote: Total 43 (delta 8), reused 42 (delta 7), pack-reused 0        \n'        
[DEBUG  ] Finished ['git', '-c', 'diff.ignoreSubmodules=none', 'clone', '--progress', 'git@github.com:adswa/windowstesting.git', '/tmp/mimimimi'] with status 0                                                                     
[DEBUG  ] Git clone completed                                         
```

I suspect whatever runs this clone call has gotten suboptimal inputs/parameters.

### PR DESCRIPTION
**Pull request by christian-monch**
Remove an unnecessary warning from the runner in the case where `stdin` is provided, but does not fit any of our specially handled cases.

Fixes #6395 

#### 🐛 Bug Fixes
- Fixes #6395   remove an unnecessary and misleading warning

### CODE REVIEW

**Comment by adswa** [comment]
Thanks for the fix! Just tried it out :+1:

**Comment by christian-monch** [comment]
> Thanks for the fix! Just tried it out +1

Your welcome :-)

**Comment by mih** [comment]
Is the `nosec` interpreted by codeclimate?

File: datalad/runner/nonasyncrunner.py [hunk 1]
```diff
@@ -304,42 +303,46 @@ def run(self) -> Union[Any, Generator]:
...
+        # and trusted. Security check is therefore skipped.
         kwargs = {
             **self.popen_kwargs,
             **dict(
                 bufsize=0,
                 stdin=subprocess.PIPE if self.write_stdin else self.stdin,
                 stdout=subprocess.PIPE if self.catch_stdout else None,
                 stderr=subprocess.PIPE if self.catch_stderr else None,
-                shell=True if isinstance(self.cmd, str) else False
+                shell=True if isinstance(self.cmd, str) else False      # nosec
```
Comment (yarikoptic) [inline datalad/runner/nonasyncrunner.py:329]: What nosec - what understand it, code climate?

**Comment by christian-monch** [comment]
> Is the nosec interpreted by codeclimate?

yes.

File: datalad/runner/nonasyncrunner.py (hunk 1 above)
Comment (christian-monch) [reply datalad/runner/nonasyncrunner.py:329]: It is understood by [bandit](https://github.com/PyCQA/bandit), a security linter that is used by codeclimate.

**Review by mih — APPROVED** [review]
LGTM, thx! I restarted the error-ing test. Once they passed, we could merge from my POV.

**Pull request merged**